pkg/controller/controller.go

```diff
@@ -758,6 +758,18 @@ func (c *Controller) cleanupOrphanedResources(mbPair *types.ModelBindingPair) {
 		glog.Errorf("Failed to cleanup custom resources for binding %s: %v", mbPair.Binding.Name, err)
 	}
 
+	// Cleanup Deployments for generic components
+	err = managed.CleanupOrphanedDeployments(mbPair, c.managedClusterConnections)
+	if err != nil {
+		glog.Errorf("Failed to cleanup deployments for binding %s: %v", mbPair.Binding.Name, err)
+	}
+
+	// Cleanup Services for generic components
+	err = managed.CleanupOrphanedServices(mbPair, c.managedClusterConnections)
```

Comment:
Shouldn't services be cleaned up before deployments?

Reply:
Cleanup was happening in the reverse order of how resources were created.  I can change the order but I don't think it matters.

Reply:
Looks like the same order, deployments are create first then services.  So, services should be deleted first then deployments.

Reply:
changed order